tests/e2e/e2e_test.go

```diff
@@ -1,63 +1,166 @@
 
 	RegisterFailHandler(Fail)
+
+	ctx = context.Background()
+	getTestEnv()
+
 	RunSpecs(t,
 		suiteName,
 	)
 }
 
 // BeforeSuite runs before the test suite
 var _ = BeforeSuite(func() {
-	ctx = context.Background()
+	runner = NewRunner(
+		WithHost(host),
+		WithPort(sshPort),
+		WithSshKey(sshKey),
+		WithSshUser(sshUser),
+	)
+
+	if installCTK {
+		installer, err := NewToolkitInstaller(
+			WithRunner(runner),
+			WithImage(ImageRepo+":"+ImageTag),
+			WithTemplate(dockerInstallTemplate),
+		)
+		Expect(err).ToNot(HaveOccurred())
+
+		err = installer.Install()
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	_, _, err := runner.Run("docker pull ubuntu")
+	Expect(err).ToNot(HaveOccurred())
+
+	_, _, err = runner.Run(fmt.Sprintf("docker pull %s", vectorAddImage))
+	Expect(err).ToNot(HaveOccurred())
+
+	_, _, err = runner.Run(fmt.Sprintf("docker pull %s", deviceQueryImage))
+	Expect(err).ToNot(HaveOccurred())
+	_, _, err = runner.Run(fmt.Sprintf("docker pull %s", cudaImage))
+	Expect(err).ToNot(HaveOccurred())
 })
+
+// getTestEnv gets the test environment variables
+func getTestEnv() {
+	defer GinkgoRecover()
+	var err error
+
+	_, thisFile, _, _ := runtime.Caller(0)
+	packagePath = filepath.Dir(thisFile)
+
+	installCTK = getBoolEnvVar("E2E_INSTALL_CTK", false)
+
+	ImageRepo = os.Getenv("E2E_IMAGE_REPO")
+	Expect(ImageRepo).NotTo(BeEmpty(), "E2E_IMAGE_REPO environment variable must be set")
+
+	ImageTag = os.Getenv("E2E_IMAGE_TAG")
+	Expect(ImageTag).NotTo(BeEmpty(), "E2E_IMAGE_TAG environment variable must be set")
+
+	// TODO (@ArangoGutierrez):
+	// once https://github.com/NVIDIA/nvidia-container-toolkit/pull/602
+	// is merged, remove this
+	ImageTag = fmt.Sprintf("%s-ubuntu20.04", ImageTag)
+
+	sshKey = os.Getenv("E2E_SSH_KEY")
+	Expect(sshKey).NotTo(BeEmpty(), "E2E_SSH_KEY environment variable must be set")
+
+	sshUser = os.Getenv("E2E_SSH_USER")
+	Expect(sshUser).NotTo(BeEmpty(), "E2E_SSH_USER environment variable must be set")
+
+	host = os.Getenv("E2E_SSH_HOST")
+	Expect(host).NotTo(BeEmpty(), "E2E_SSH_HOST environment variable must be set")
+
+	sshPort = getIntEnvVar("E2E_SSH_PORT", 22)
+
+	// Get current working directory
+	cwd, err = os.Getwd()
+	Expect(err).NotTo(HaveOccurred())
+}
+
+// getBoolEnvVar returns the boolean value of the environment variable or the default value if not set.
+func getBoolEnvVar(key string, defaultValue bool) bool {
+	value := os.Getenv(key)
+	if value == "" {
+		return defaultValue
+	}
+	boolValue, err := strconv.ParseBool(value)
+	if err != nil {
+		return defaultValue
+	}
+	return boolValue
+}
+
+// getIntEnvVar returns the integer value of the environment variable or the default value if not set.
+func getIntEnvVar(key string, defaultValue int) string {
+	value := os.Getenv(key)
+	if value == "" {
+		return strconv.Itoa(defaultValue)
+	}
+	intValue, err := strconv.Atoi(value)
+	if err != nil {
+		return strconv.Itoa(defaultValue)
+	}
+
+	return strconv.Itoa(intValue)
```

Comment:
We're calling `Itoa` three times here -- twice with the same argument -- is there not a better way to implement this. What about splitting this into two functions:

1. `getEnvVarAs[T](string) (T, error)`
2. `getEnvVarOrDefault[T](string, T) T`

we could then call `getEnvVarAs` from `getEnvVarOrDefault` and return the default value if we return an error.